DESIGN.md

```diff
@@ -0,0 +1,27 @@
+## Overview
+
+### Target Audience
+
+- Developer A: high-level framework writer
+  - wants to provide barebones initial configuration
+  - wants to consume a simple event stream
+  - wants to inject events into a managed event loop (easy state machine semantics)
+
+```go
+c := client.New(connectionOptions..) // package scheduler/client
+eventCh, errCh := c.Register(registrationOptions..)
```

Comment:
I am not sure of libraries (in this case `mesos-go`) returning channels. We should leave concurrency handling to the consumers of the library.

What I'd rather suggest is to leave mesos-go blocking as much as possible, like the `net/http` client package and simply have a blocking call like
```
event, err := c.Event()
```